lib/puppet/configurer.rb

```diff
@@ -236,6 +262,7 @@ def run_internal(options)
       return nil
     ensure
       execute_postrun_command or return nil
+      terminate_http_pool_monitoring(Puppet.lookup(:http_pool))
```

Comment:
If the postrun command fails, e.g. path is incorrect, then `execute_postrun_command` will return false and we will never call `terminate_http_pool_monitoring`